homeassistant/components/folder_watcher.py

```diff
@@ -0,0 +1,85 @@
+"""
+Component for monitoring activity on a folder.
+"""
+import os
+import logging
+import voluptuous as vol
+from watchdog.observers import Observer
+from watchdog.events import FileSystemEventHandler
+
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['watchdog']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_FOLDERS = 'folders'
+DOMAIN = "folder_watcher"
+DEFAULT_IS_RECURSIVE = True
+EVENT_TYPE = "event_type"
+IS_DIRECTORY = "is_directory"
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_FOLDERS):
+            vol.All(cv.ensure_list, [cv.isdir]),
+        })
```

Comment:
Could add config for file types to watch/not_watch - should this be applied to all folders or folder by folder? I'm not a fan of overly complex config

Reply:
I think if this should be added, then as a [glob-style](https://docs.python.org/3/library/glob.html) filter, though I don't think the it's really something we should do anyway.

Reply:
@OttoWinter what is your thinking behind not adding?